app/src/main/res/layout/activity_entry_name.xml

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/activity_entry_name"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:fitsSystemWindows="true"
+    android:paddingBottom="@dimen/activity_vertical_margin"
+    android:paddingLeft="@dimen/activity_horizontal_margin"
+    android:paddingRight="@dimen/activity_horizontal_margin"
+    android:paddingTop="@dimen/activity_vertical_margin"
+    tools:context="io.intrepid.contest.screens.entrysubmission.entryname.EntryNameActivity"
+    >
+
+    <TextView
+        android:id="@+id/contestant_welcome_text_view"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        />
+
+    <io.intrepid.contest.customviews.ClearableEditText
+        android:id="@+id/entry_name_edit_text"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_below="@id/contestant_welcome_text_view"
+        android:hint="@string/name_entry"
+        />
+
+    <Button
+        android:id="@+id/entry_name_next_button"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_alignParentBottom="true"
+        android:enabled="false"
+        android:text="@string/common_next"
+        android:visibility="gone"
```

Comment:
I recommend adding tools:visibility="visible", it doesnt affect the code but then will show the button in the view window of android studio while you're debugging.